src/modules/pki/field.rs

```diff
@@ -0,0 +1,159 @@
+use std::{
+    collections::HashMap,
+    sync::Arc,
+};
+
+use crate::{
+    logical::{Field, FieldType},
+    new_fields, new_fields_internal,
+};
+
+pub fn ca_common_fields() -> HashMap<String, Arc<Field>> {
+    let fields = new_fields!({
+        "alt_names": {
+            field_type: FieldType::Str,
+            required: false,
+            description: r#"The requested Subject Alternative Names, if any,
+in a comma-delimited list. May contain both DNS names and email addresses."#
+        },
+        "common_name": {
+            field_type: FieldType::Str,
+            required: true,
+            description: r#"The requested common name; if you want more than
+one, specify the alternative names in the alt_names map. If not specified when
+signing, the common name will be taken from the CSR; other names must still be
+specified in alt_names or ip_sans.
+"#
+        },
+        "ttl": {
+            field_type: FieldType::Str,
+            description: r#"The requested Time To Live for the certificate;
+sets the expiration date. If not specified the role default, backend default,
```

Comment:
What is the difference between TTL and expiration date?

Reply:
> What is the difference between TTL and expiration date?

Prioritize using the 'not_after' parameter; if not configured, use 'ttl'.